.github/workflows/scripts/get-preview-features.js

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env node
+// @ts-check
+
+const fs = require('fs')
+const { getPreviewFeatures } = require('$internals')
+
+async function main() {
+  /** @type string[] */
+  let previewFeatures
+  const featureSet = process.env.PREVIEW_FEATURE_SET ?? 'none'
+  const included = (process.env.INCLUDED_PREVIEW_FEATURES ?? '').split(',')
+  const excluded = (process.env.EXCLUDED_PREVIEW_FEATURES ?? '').split(',')
+
+  previewFeatures = featureSet === 'all' ? getPreviewFeatures() : included
+  previewFeatures = previewFeatures.filter((pf) => !excluded.includes(pf))
+
+  const commaSeparatedPreviewFeaturesString = previewFeatures.join(',')
+
+  if (typeof process.env.GITHUB_OUTPUT == 'string' && process.env.GITHUB_OUTPUT.length > 0) {
+    fs.appendFileSync(process.env.GITHUB_OUTPUT, `previewFeatures=${commaSeparatedPreviewFeaturesString}\n`)
+    console.debug(`${commaSeparatedPreviewFeaturesString} added to GITHUB_OUTPUT`)
+  }
+}
```

Comment:
Uses `@prisma/internals` to get the current engine's full list of preview features.

If the feature set is set to `full`, you get the full list of features minus the exclusions.
Additionally, it's possible to force the inclusion of a preview feature so it's always active regardless of the feature set.

Reply:
Ah nice, this looks a lot cleaner than what we started with 😁